rerun_cpp/src/rerun/timeline.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "error.hpp"
+
+#include <string>
+
+struct rr_timeline;
+
+namespace rerun {
+    /// Describes the type of a timeline or time point.
+    enum class TimeType {
+        Time = 0,
+        Sequence = 1,
+    };
+
+    /// Definition of a timeline.
+    struct Timeline {
+        /// The name of the timeline.
+        std::string name;
+
+        /// The type of the timeline.
+        TimeType type;
+
+      public:
+        /// Creates a new timeline.
+        Timeline(std::string_view _name, TimeType _type = TimeType::Time)
+            : name(std::string(_name)), type(_type) {}
```

Comment:
this will avoid a copy if the user passes a `std::string` in.

Also, see https://github.com/rerun-io/rerun/blob/main/CODE_STYLE.md#members

```suggestion
        Timeline(std::string name_, TimeType type_ = TimeType::Time)
            : name(std::move(name_)), type(type_) {}
```